packages/contracts/contracts/factory/PluginRepoFactory.sol

```diff
@@ -32,29 +33,32 @@ contract PluginRepoFactory {
     /// @param _name The plugin repository name.
     /// @param _initialOwner The plugin maintainer address.
     /// TODO: Rethink if it need permission to prevent it from getting poluted, same for `createPluginRepoWithFirstVersion`.
-    function createPluginRepo(string calldata _name, address _initialOwner)
-        external
-        returns (PluginRepo)
-    {
+    function createPluginRepo(
+        string calldata _name,
+        address _initialOwner
+    ) external returns (PluginRepo) {
         return _createPluginRepo(_name, _initialOwner);
     }
 
     /// @notice Creates and registers a named `PluginRepo` and publishes an initial version.
     /// @dev The initial owner of the new PluginRepo is `address(this)`, afterward ownership will be transfered to the address `_maintainer`.
     /// @param _name The plugin repository name.
     /// @param _pluginSetup The plugin factory contract associated with the plugin version.
-    /// @param _contentURI The external URI for fetching the new version's content.
     /// @param _maintainer The plugin maintainer address.
+    /// @param _releaseMetadata The external URI for fetching the new version's release content.
+    /// @param _buildMetadata The external URI for fetching the new version's build content.
     function createPluginRepoWithFirstVersion(
         string calldata _name,
         address _pluginSetup,
-        bytes memory _contentURI,
-        address _maintainer
+        address _maintainer,
+        bytes memory _releaseMetadata,
+        bytes memory _buildMetadata
     ) external returns (PluginRepo pluginRepo) {
         // Sets `address(this)` as initial owner which is later replaced with the maintainer address.
         pluginRepo = _createPluginRepo(_name, address(this));
```

Comment:
After the ``createPluginRepoWithFirstVersion` call is done, we revoke `CREATE_VERSION_PERMISSION_ID` for factory, so it can no longer do anything on plugin repo.